lighthouse-core/gather/driver.js

```diff
@@ -942,55 +940,21 @@ class Driver {
    * @return {Promise<LH.Trace>}
    */
   endTrace() {
-    return new Promise((resolve, reject) => {
-      // When the tracing has ended this will fire with a stream handle.
-      this.once('Tracing.tracingComplete', completeEvent => {
-        this._readTraceFromStream(completeEvent)
-            .then(traceContents => resolve(traceContents), reject);
-      });
+    /** @type {Array<LH.TraceEvent>} */
+    const traceEvents = [];
 
-      // Issue the command to stop tracing.
-      return this.sendCommand('Tracing.end').catch(reject);
+    // setup listener for when a trace bundle is sent
```

Comment:
"trace chunk" but yeah. How about "// dataCollected events fire for each trace chunk, in order"

Reply:
Changed, see if u like it.